### PR TITLE
add optimizer's def funcs and attributres to save/load

### PR DESCRIPTION
**Pull request by EdwardTyantov**
to access param_groups, state_dict in standart way

### CODE REVIEW

**Comment by EdwardTyantov** [comment]
Hi,
Answering your question - Yes, first of all for saving/loading, then as you pointed out you should incorporate the global state into state_dict. 

As mentioned in the doc this optimizer can be used instead of sgd or adam seamlessly. Many pytorchers use lr schedulers, i.e. https://github.com/pytorch/pytorch/blob/630af4d7d8812efa43a01d938f0fc53c47802b0b/torch/optim/lr_scheduler.py . They adapt learning rate using these methods. In your example code you use some kind of scheduling too https://github.com/JianGoForIt/YellowFin_Pytorch/blob/master/pytorch-cifar/main.py#L111 . 

This pull request obv will not work as I understand now ;)

**Comment by JianGoForIt** [comment]
closed as already addressed in other pr and commits.
